### PR TITLE
Wrappers

### PR DESCRIPTION
**Pull request by farizrahman4u**
Base class for Wrappers.
Functions that should be re-implemented in derived classes:

* `get_input_shape()` : Function to infer the wrapper's input shape from the child layer's input shape
* `get_child_input_shape()` : Inverse of the above function. Infer the child's input shape from the wrapper's shape.
* `get_output_shape()` : Function to infer the wrapper's output shape from the child layer's output  shape
* `get_output()`

Does not work for multiple child layers (Such as in Bidirectional RNN, where 2 RNNs have to wrapped into a single layer)

### CODE REVIEW

File: keras/layers/wrappers.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        return self.input_shape
+
+    def get_output_shape(self):
+        '''
+        Get output shape of the wrapper from the child layer's output shape
+        '''
+        return self.layer.output_shape
+
+    @property
+    def trainable_weights(self):
```
Comment (fchollet) [inline keras/layers/wrappers.py]: For the record, you cannot set this as a `property` because `core.Layer` will attempt to set it as a class attribute.

Comment (farizrahman4u) [inline keras/layers/wrappers.py]: Setters have been added.

**Comment by neggert** [comment]
What's the use case here? Not criticizing, just curious.

**Comment by farizrahman4u** [comment]
@neggert  #1827

**Comment by fchollet** [comment]
Looks good to me. Please squash, I will merge, then add the `TimeDistributed` layer on top of it (unless you want to take care of `TimeDistributed` too).

**Comment by farizrahman4u** [comment]
@fchollet Done.

File: keras/layers/wrappers.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        return (layer_input_shape[0], timesteps) + layer_input_shape[1:]
+
+    def get_child_input_shape(self):
+        input_shape = list(self.input_shape)
+        input_shape.pop(1)
+        return input_shape
+
+def get_output(self, train=False):
+    X = self.get_input(train)  # (nb_samples, timesteps, ..., input_dim)
+    if self.layer.input_sahpe[0]:
```
Comment (gvessere) [inline keras/layers/wrappers.py]: typo

**Comment by neggert** [comment]
@farizrahman4u @fchollet What's status on this PR? I'm looking at implementing a residual layer and it looks like it would be trivial if I could inherit from the `Wrapper` class in this PR.

**Comment by mossaab0** [comment]
I also really need this commit to be merged, as none of the alternatives for building a stacked BLSTM work for me currently (i.e., merging left and right sequential layers, and using graph models).

**Comment by fchollet** [comment]
Just wait for Keras 1.0 (to be released soon).

**Comment by fchollet** [comment]
Closing outdated PRs following the release of Keras 1.0.

**Pull request closed without merging**